src/Controller/Api/TwoFactorAuthController.php

```diff
@@ -11,14 +11,14 @@
 use Nelmio\ApiDocBundle\Annotation\Model;
 use Scheb\TwoFactorBundle\Security\TwoFactor\Provider\Google\GoogleAuthenticatorInterface;
 use Swagger\Annotations as SWG;
-use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
 
-final class TwoFactorAuthController extends Controller
+class TwoFactorAuthController extends AbstractController
```

Comment:
I am not familiar with this practice. Why put `final` everywhere?

Reply:
Because I know for sure that this controller will not be reused. 

It documents that a class was not designed with inheritance in mind, and this is usually true for all classes by default.

Reply:
Why then the `final` is not included in the class definition by design of lang?